### PR TITLE
Nicolas.laurenti/calculator improvements

### PR DESCRIPTION
**Pull request by jnicolaslaurenti**
In this PR, I added improvements in the calculator

Improvements:

* Division for zero
* Two operator in a row(now, the second operator replace the first)
* Press the equal button when the first operand is empty
* Press the equal button when the second operand is empty
* Press the equal button when the operator is empty
* Press a operator button when the first operand is empty

Pixel 3A:
Calculator_improvements_Pixel3A

### CODE REVIEW

**Review by BQuinteros — APPROVED** [review]

File: app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java [hunk 1]
```diff
@@ -28,19 +30,28 @@ public void showError() {
...
     @Override
     public void showDeleteMessage() {
         Context context = getContext();
         if (context != null) {
-            Toast.makeText(context, context.getString(R.string.value_deleted), Toast.LENGTH_LONG).show();
+            Toast.makeText(context, context.getString(R.string.value_deleted), Toast.LENGTH_SHORT).show();
         }
     }
+
+    public void reset() {
```
Comment (mauroLonghi) [inline app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java]: pls change the name of the method to resetVisor or resetResultVisor or else

Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java]: Solved!

File: app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java [hunk 2]
```diff
@@ -28,19 +30,28 @@ public void showError() {
...
     public void showDeleteMessage() {
         Context context = getContext();
         if (context != null) {
-            Toast.makeText(context, context.getString(R.string.value_deleted), Toast.LENGTH_LONG).show();
+            Toast.makeText(context, context.getString(R.string.value_deleted), Toast.LENGTH_SHORT).show();
         }
     }
+
+    public void resetResultView() {
+        binding.textViewResult.setText(RESULT);
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java]: you are in the View, so you can access the String res, do not use the constant RESULT here

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java (hunk 2 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/calculator/mvp/view/CalculatorView.java]: Solved!

**Review by SartiMau — APPROVED** [review]

**Review by benjaglobant — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Pull request merged**